### PR TITLE
Allow config env to set autoload path

### LINKED ISSUE #37331: Config autoload_paths cannot be modified by environment configurations
### Steps to reproduce
1. Configure environment eg. "test" to add a new autoload path

### Expected behavior
1. Autoload paths is changed for the environment 
I was assuming autoload paths can be changed based on environment as that would be consistent with other `config`

### Actual behavior
1. Autoload paths is not changed as it is already frozen
2. Even if you duplicate the autoload paths it is not respected because ActiveSupport::Dependencies.autoload_paths has already been set, you can modify this yourself but the guide says not to do so:
```
Please, do not mutate ActiveSupport::Dependencies.autoload_paths, the public interface to change autoload paths is config.autoload_paths
```

### System configuration
Rails: Master as of 57ae917
Ruby: 2.6

### Preliminary Investigation
- `set_load_paths` and `set_autoload_paths` happens before the `load_environment_config`
https://github.com/rails/rails/blob/57ae917e9d52c9472fd37adbfa47d6de93dd01bf/railties/lib/rails/engine.rb#L575-L615
- moving the `set_load_paths` `set_autoload_paths` after the environment seems to work
  - Possibly makes sense since with Zeitwerk autoloading in initializers is deprecated, you receive the warning: `DEPRECATION WARNING: Initialization autoloaded the constants`

### PR DESCRIPTION
**Pull request by aldhsu**
### Summary
Resolves https://github.com/rails/rails/issues/37331

Moves environment file loading before autoload path initialization.
Attention @fxn.

### CODE REVIEW

File: guides/source/autoloading_and_reloading_constants_classic_mode.md [hunk 1]
```diff
@@ -482,8 +482,6 @@ What is described above are the defaults with a newly generated Rails app. There
 
 See also [Autoloading in the Test Environment](#autoloading-in-the-test-environment).
 
-`config.autoload_paths` is not changeable from environment-specific configuration files.
-
```
Comment (aldhsu) [inline guides/source/autoloading_and_reloading_constants_classic_mode.md]: I couldn't find a reason this was the case?

Comment (fxn) [reply guides/source/autoloading_and_reloading_constants_classic_mode.md]: Probably the docs were describing how things worked. I'd bet this is an overlook and just a consequence of how the initializers were ordered.

**Comment by fxn** [comment]
Hey, I have worked on the tests a bit and have this ready.

It is too late for 6.0.1, because there is a RC out, I'll push after the release.

**Comment by fxn** [comment]
In `6-0-stable` (dede40ecadf632f72a1143396043a1994e65c839) now, and also cherry-picked to `master`.

Thanks!

**Pull request closed without merging**